src/Form/Type/FormatterType.php

```diff
@@ -63,17 +64,42 @@ final class FormatterType extends AbstractType
      */
     private $toolbarManager;
 
+    /**
+     * NEXT_MAJOR: Change signature for
+     * (PoolInterface $pool, TranslatorInterface $translator, CKEditorConfigurationInterface $ckEditorConfiguration).
+     *
+     * @param ConfigManagerInterface|CKEditorConfigurationInterface $configManagerOrCkEditorConfiguration
+     */
     public function __construct(
         PoolInterface $pool,
         TranslatorInterface $translator,
-        ConfigManagerInterface $configManager,
-        PluginManagerInterface $pluginManager,
-        TemplateManagerInterface $templateManager,
-        ToolbarManagerInterface $toolbarManager
+        object $configManagerOrCkEditorConfiguration,
+        ?PluginManagerInterface $pluginManager = null,
+        ?TemplateManagerInterface $templateManager = null,
+        ?ToolbarManagerInterface $toolbarManager = null
     ) {
+        if (!$configManagerOrCkEditorConfiguration instanceof CKEditorConfigurationInterface) {
+            @trigger_error(sprintf(
+                'Passing %s as argument 3 to %s() is deprecated since sonata-project/formatter-bundle 4.x'
+                .' and will throw a \TypeError in version 5.0. You must pass an instance of %s instead.',
+                \get_class($configManagerOrCkEditorConfiguration),
+                __METHOD__,
+                CKEditorConfigurationInterface::class
+            ), E_USER_DEPRECATED);
+
+            if (!$configManagerOrCkEditorConfiguration instanceof ConfigManagerInterface) {
+                throw new \TypeError(sprintf(
+                    'Argument 3 passed to %s() must be an instance of %s, %s given.',
+                    __METHOD__,
+                    ConfigManagerInterface::class,
+                    \get_class($configManagerOrCkEditorConfiguration)
+                ));
+            }
+        }
+
         $this->pool = $pool;
         $this->translator = $translator;
-        $this->configManager = $configManager;
+        $this->configManagerOrCkEditorConfiguration = $configManagerOrCkEditorConfiguration;
```

Comment:
```suggestion
        $this->configManager = $configManagerOrCkEditorConfiguration;
        $this->configManagerOrCkEditorConfiguration = $configManagerOrCkEditorConfiguration;
```
Or you can set it in if clausule.